packages/hydrogen/src/storefront.ts

```diff
@@ -149,26 +148,25 @@ export function createStorefrontClient<TI18n extends I18nBase>({
     );
   }
 
-  const {
-    getPublicTokenHeaders,
-    getPrivateTokenHeaders,
-    getStorefrontApiUrl,
-    getShopifyDomain,
-  } = createStorefrontUtilities(clientOptions);
-
-  const getHeaders = clientOptions.privateStorefrontToken
-    ? getPrivateTokenHeaders
-    : getPublicTokenHeaders;
-
-  const defaultHeaders = getHeaders({contentType: 'json'});
+  const authorizer = createAuthorizer({
+    storeDomain: clientOptions.storeDomain,
+    storefrontAPIVersion: clientOptions.storefrontApiVersion,
+    publicAccessToken: clientOptions.publicStorefrontToken,
+    privateAccessToken: clientOptions.privateStorefrontToken,
+    contentType: clientOptions.contentType,
+    sdkVariant: 'Hydrogen',
+    sdkVariantSource: 'react',
+    sdkVersion: clientOptions.storefrontApiVersion,
+  });
+
+  const defaultHeaders = authorizer.getApiHeaders({
+    tokenType: clientOptions.privateStorefrontToken ? 'private' : 'public',
```

Comment:
Since both `private` and `public` tokens can be added to the authorizer, we've opted to have the devs explicitly tell the authorizer which token they'd like to use to generate the headers for clarity. 

However, even if we do set a default, I would opt to set the default to `public` since it's less risky to use the `public` token than to accidentally expose the `private` token. 

Taking into consideration with hydrogen's use case where `private` token is likely more used than the `public` token, we can introduce a `defaultHeaderTokenType` (defaults to `public`) that's set at initialization where devs can choose which token type they'd like to use by default when calling `getApiHeaders()`.

Reply:
All the previous points are good 🙌  

Counter argument, if we default to `publicToken`, I imagine folks not realizing they have to explicitly set to use the `privateToken` and then receiving rate limiting errors, then contacting us (which happened often in hydrogen).

Another idea, how about having typescript and error checking help us here? we only allow at client creation time either `private` or `public` token, not both. This could signal devs to be careful on choosing the right token for the right setting.

Reply:
The client in the library actually only supports a single token and we default the token type to `public` (although they can be set to `private` at initialization) since we do expect most of our users to use the client in the browser. 

For the authorizer, it accepts both `private` and `public` tokens because the `hydrogen-ui` that I was referring to at the time, allowed for it - if hydrogen will generally support a single token type only, I think having the authorizer support a single token type makes sense and I do like that the devs will have to make a explicit choice on which token to use at initialization.